config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/Http2AssertionBuilder.java

```diff
@@ -0,0 +1,39 @@
+package com.ca.apim.gateway.cagatewayconfig.bundle.builder;
+
+import com.ca.apim.gateway.cagatewayconfig.beans.Bundle;
+import com.ca.apim.gateway.cagatewayconfig.beans.GenericEntity;
+import com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentParseException;
+import org.w3c.dom.Element;
+
+import static com.ca.apim.gateway.cagatewayconfig.bundle.builder.EntityBuilder.BundleType.ENVIRONMENT;
+import static com.ca.apim.gateway.cagatewayconfig.util.policy.PolicyXMLElements.*;
+import static com.ca.apim.gateway.cagatewayconfig.util.policy.PolicyXMLElements.GOID_VALUE;
+import static com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentUtils.createElementWithAttribute;
+import static com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentUtils.getSingleChildElement;
+
+public class Http2AssertionBuilder implements PolicyAssertionBuilder {
+    @Override
+    public void buildAssertionElement(Element assertionElement, PolicyBuilderContext policyBuilderContext) {
+        final Bundle bundle = policyBuilderContext.getBundle();
+        final Element http2ClientNameEle = getSingleChildElement(assertionElement, HTTP2_CLIENT_CONFIG_NAME, true);
+        if (http2ClientNameEle != null) {
```

Comment:
We could have gone with the XPATH way to minimize or reduce the redundant code.
So that, code becomes declarative and easily manageable.

Reply:
Yeah that would be better but we have to refactor the code completely in all the assertions, may be we can consider this in the next sprint